Extension/src/LanguageServer/configurations.ts

```diff
@@ -396,7 +396,15 @@ export class CppProperties {
             (isUnset(settings.defaultCompileCommands) || settings.defaultCompileCommands === "") && !configuration.compileCommands) {
             // compile_commands.json already specifies a compiler. compilerPath overrides the compile_commands.json compiler so
             // don't set a default when compileCommands is in use.
-            configuration.compilerPath = this.defaultCompilerPath;
+
+            // if the compiler is a cl.exe compiler, replace the full path with the "cl.exe" string.
+            const compiler: string = path.basename(this.defaultCompilerPath).toLowerCase();
+
+            configuration.compilerPath = "cl.exe";
```

Comment:
Did `configuration.compilerPath = "cl.exe";` get moved out of the conditional block?